plugin.video.vtm.go/resources/lib/modules/library.py

```diff
@@ -0,0 +1,195 @@
+        _LOGGER.debug('Checking if movie %s is still available', movie)
+
+        # Our parent path always exists
+        if movie is None:
+            kodiutils.library_return_status(True)
+            return
+
+        if kodiutils.get_setting_int('library_movies') == LIBRARY_FULL_CATALOG:
+            id_list = self._api.get_catalog_ids()
+        else:
+            id_list = self._api.get_mylist_ids()
+
+        kodiutils.library_return_status(movie in id_list)
+
+    def check_library_tvshow(self, program):
+        """ Check if the given program is still available. """
+        _LOGGER.debug('Checking if program %s is still available', program)
+
+        # Our parent path always exists
+        if program is None:
+            kodiutils.library_return_status(True)
+            return
+
+        if kodiutils.get_setting_int('library_tvshows') == LIBRARY_FULL_CATALOG:
+            id_list = self._api.get_catalog_ids()
+        else:
+            id_list = self._api.get_mylist_ids()
+
+        kodiutils.library_return_status(program in id_list)
+
+    @staticmethod
+    def mylist_added(video_type, content_id):
+        """ Something has been added to My List. We want to index this. """
+        if video_type == CONTENT_TYPE_MOVIE:
+            if kodiutils.get_setting_int('library_movies') != LIBRARY_ONLY_MYLIST:
+                return
+            # This unfortunately adds the movie to the database with the wrong parent path:
+            # Library().update('plugin://plugin.video.vtm.go/library/movies/?movie=%s&kodi_action=refresh_info' % content_id)
+            Library().update('plugin://plugin.video.vtm.go/library/movies/')
+
+        elif video_type == CONTENT_TYPE_PROGRAM:
+            if kodiutils.get_setting_int('library_tvshows') != LIBRARY_ONLY_MYLIST:
+                return
+            Library().update('plugin://plugin.video.vtm.go/library/tvshows/?program=%s&kodi_action=refresh_info' % content_id)
+
+    @staticmethod
+    def mylist_removed(video_type, content_id):
+        """ Something has been removed from My List. We want to de-index this. """
+        if video_type == CONTENT_TYPE_MOVIE:
+            if kodiutils.get_setting_int('library_movies') != LIBRARY_ONLY_MYLIST:
+                return
+            Library().clean('plugin://plugin.video.vtm.go/library/movies/?movie=%s' % content_id)
+
+        elif video_type == CONTENT_TYPE_PROGRAM:
+            if kodiutils.get_setting_int('library_tvshows') != LIBRARY_ONLY_MYLIST:
+                return
+            Library().clean('plugin://plugin.video.vtm.go/library/tvshows/?program=%s' % content_id)
+
+    @staticmethod
+    def configure():
+        """ Configure the library integration. """
+        # There seems to be no way to add sources automatically.
+        # * https://forum.kodi.tv/showthread.php?tid=228840
+
+        # Open the sources view
+        kodiutils.execute_builtin('ActivateWindow(Videos,sources://video/)')
+
+    @staticmethod
```

Comment:
I know we don't have any better option to support addon content in the library at the moment. I hope that you guys will remove all these code/hacks when (and if) montellese's library integration features are supported.

Reply:
I would love to do that, it took me quite some time to figure out a way that works.